apps/src/blockly/addons/cdoXml.js

```diff
@@ -165,6 +166,44 @@ export function addMutationToBehaviorDefBlocks(blockElement) {
   }
 }
 
+/**
+ *
+ * @param {Element} blockElement - The XML element for a single block.
+ */
+export function addNameToBlockFunctionDefinitionBlock(blockElement) {
+  const blockType = blockElement.getAttribute('type');
+  if (blockType !== 'procedures_defnoreturn') {
+    return;
+  }
+  const fieldElement = blockElement.querySelector('field[name="NAME"]');
+  if (!fieldElement) {
+    return;
+  }
+
+  if (fieldElement.textContent === '') {
+    fieldElement.textContent = 'unnamed';
```

Comment:
do we have to worry about translations or does blockly always use `'unnamed'`?

Reply:
This matches our current behavior for students who rename existing functions. Did we already have a task to localize the default "do something" name as well?

Reply:
I see! I was still seeing English when I switched languages but that must just be because the translations haven't been added yet.